### ISSUE TITLE
[JENKINS-20327] “Form too large” errors submitting view configurations with many jobs

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#7128]
<p>I have created a new job (by copying an older job), and saved.<br/>
I want to integrate the new job into a Jenkins view, select it, press "Edit view", select the new job, press save</p>

<p>and get the following Java exception:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-none">javax.servlet.ServletException: java.lang.IllegalStateException: Form too large 241320>200000
  at …
</pre>
</div></div>

<p>The issue is critical, because I have to implement a lot of new jobs for our testing environment, we are short before a new release of our software.</p>

<p>Thanks a lot for providing a solution for this problem.</p>

<p>Attached is the above stacktrace, as well as the output found in the error logging on the Jenkins master server, and the version of all installed plugins.</p>

**Attachments:**

- [JavaException-FormTooLarge.docx](https://issues.jenkins.io/secure/attachment/24614/JavaException-FormTooLarge.docx)

---
<details><summary><i>Originally reported by <a title="Angela Bauer" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=baue_ang">baue_ang</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-20327" target="_blank">“Form too large” errors submitting view configurations with many jobs</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Stephen Connolly" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=stephenconnolly">stephenconnolly</a>
<li><b>reported by</b>: baue_ang
<li><b>status</b>: Closed
<li><b>priority</b>: Critical
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2015-11-24T18:13:12+00:00
<li><b>votes</b>: 11
<li><b>watchers</b>: 35
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vsayer">vsayer</a>:</i>
<p>I see this at Jenkins 1.542, Gerrit 2.5.2, Gerrit Trigger 2.10.1, when using the 'Query and Trigger Gerrit Patches' feature to manually trigger patchsets. Here's the stack trace:</p>

<p>java.lang.IllegalStateException: Form too large 216327>200000<br/>
	at org.eclipse.jetty.server.Request.extractParameters(Request.java:352)<br/>
	at org.eclipse.jetty.server.Request.getParameter(Request.java:790)<br/>
	at javax.servlet.ServletRequestWrapper.getParameter(ServletRequestWrapper.java:184)<br/>
	at org.kohsuke.stapler.QueryParameter$HandlerImpl.parse(QueryParameter.java:69)<br/>
	at org.kohsuke.stapler.QueryParameter$HandlerImpl.parse(QueryParameter.java:62)<br/>
	at org.kohsuke.stapler.AnnotationHandler.handle(AnnotationHandler.java:91)<br/>
	at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:153)<br/>
	at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:96)<br/>
	at org.kohsuke.stapler.MetaClass$1.doDispatch(MetaClass.java:120)<br/>
	at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:53)<br/>
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:728)<br/>
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:858)<br/>
	at org.kohsuke.stapler.MetaClass$12.dispatch(MetaClass.java:390)<br/>
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:728)<br/>
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:858)<br/>
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:631)<br/>
	at org.kohsuke.stapler.Stapler.service(Stapler.java:225)<br/>
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)<br/>
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:686)<br/>
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1494)<br/>
	at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:96)<br/>
	at hudson.plugins.greenballs.GreenBallFilter.doFilter(GreenBallFilter.java:58)<br/>
	at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:99)<br/>
	at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:88)<br/>
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1482)<br/>
	at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:48)<br/>
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1482)<br/>
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)<br/>
	at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:76)<br/>
	at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:164)<br/>
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1482)<br/>
	at org.kohsuke.stapler.compression.CompressionFilter.doFilter(CompressionFilter.java:46)<br/>
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1482)<br/>
	at hudson.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:81)<br/>
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1474)<br/>
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:499)<br/>
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:137)<br/>
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:533)<br/>
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:231)<br/>
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1086)<br/>
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:428)<br/>
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:193)<br/>
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1020)<br/>
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)<br/>
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:116)<br/>
	at org.eclipse.jetty.server.Server.handle(Server.java:370)<br/>
	at org.eclipse.jetty.server.AbstractHttpConnection.handleRequest(AbstractHttpConnection.java:489)<br/>
	at org.eclipse.jetty.server.AbstractHttpConnection.content(AbstractHttpConnection.java:960)<br/>
	at org.eclipse.jetty.server.AbstractHttpConnection$RequestHandler.content(AbstractHttpConnection.java:1021)<br/>
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:865)<br/>
	at org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:240)<br/>
	at org.eclipse.jetty.server.AsyncHttpConnection.handle(AsyncHttpConnection.java:82)<br/>
	at org.eclipse.jetty.io.nio.SelectChannelEndPoint.handle(SelectChannelEndPoint.java:668)<br/>
	at org.eclipse.jetty.io.nio.SelectChannelEndPoint$1.run(SelectChannelEndPoint.java:52)<br/>
	at winstone.BoundedExecutorService$1.run(BoundedExecutorService.java:77)<br/>
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1146)<br/>
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)<br/>
	at java.lang.Thread.run(Thread.java:701)</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=baue_ang">baue_ang</a>:</i>
<p>@Vivek<br/>
please use the workaround as described by Dennys (thank you Dennys for providing it): -Dorg.eclipse.jetty.server.Request.maxFormContentSize=5000000<br/>
it worked for me, I have upgraded last week to 1.542 and got the below error again.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mwebber">mwebber</a>:</i>
<p>I got this problem in 1.547 when I edited a large view that was constructed using the <a href="https://wiki.jenkins-ci.org/display/JENKINS/Sectioned+View+Plugin" class="external-link" target="_blank" rel="nofollow noopener">Sectioned View</a> plugin.</p>

<p>I have a lot of jobs, and the way the edit form is constructed means that I hit up against the limit. This didn't use to happen - at what point was the limit introduced? I am about to try the <tt>-Dorg.eclipse.jetty.server.Request.maxFormContentSize=5000000</tt> option to see if that works.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mwebber">mwebber</a>:</i>
<p>Adding <tt>-Dorg.eclipse.jetty.server.Request.maxFormContentSize=5000000</tt> option to the Jenkins startup options does indeed fix this problem for me.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=johnrjackson">johnrjackson</a>:</i>
<p>After adding this fix to the Jenkins start-up options, the "Restrict where this project can be run" job feature is broken. Jobs fail if this feature is used with the following error:<br/>
<span class="error">&#91;EnvInject&#93;</span> - Loading node environment variables.<br/>
ERROR: SEVERE ERROR occurs<br/>
org.jenkinsci.lib.envinject.EnvInjectException: hudson.remoting.ChannelClosedException: channel is already closed</p>

<p>Is anyone else seeing the same behavior?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tautschnig">tautschnig</a>:</i>
<p>This is, perhaps unsurprisingly, and issue now present in the LTS. Given my setup with a large number of jobs, it seems I cannot create/configure a view anymore.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=davidruhmann">davidruhmann</a>:</i>
<p>I too experienced this issue using the native windows jenkins package 1.653.  However, I was not having this issue on another server running the same version of Jenkins but running on Tomcat.  Once I switched to Tomcat, the issue has disappeared.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Reducing severity as there seems to be a workaround in the System property <tt>org.eclipse.jetty.server.Request.maxFormContentSize</tt>.</p>

<p>Given the version ranges mentioned, and <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mwebber" class="user-hover" rel="mwebber">mwebber</a>'s comments, it's safe to assume this was introduced in 1.535 with the switch from Winstone to Jetty.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hirre">hirre</a>:</i>
<p>Mine hangs at e.g. 64%, but no error in log... -Dorg.eclipse.jetty.server.Request.maxFormContentSize=5000000 doesn't work either...</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jasonw">jasonw</a>:</i>
<p>For all those that the -Dorg.eclipse.jetty.server.Request.maxFormContentSize=5000000 doesn't work, add it straight after the java command and before the -jar<br/>
Example: java -Dorg.eclipse.jetty.server.Request.maxFormContentSize=5000000 -jar ${JENKINS_HOME}/jenkins.war</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick">jglick</a>:</i>
<p>Commenters: please stop pasting long stack traces into comments, as it makes JIRA pages hard to load. Use file attachments. In this case the stack trace is of no interest anyway; what counts is the contents of the form being submitted. (Which would <em>definitely</em> have to go into a file attachment since it is huge.)</p>

<p>Restricting this issue to the problem encountered when submitting a view configuration with a large number of jobs displayed as options. There are apparently less common problems with the same symptom but different causes from submitting other big forms, which would best be filed separately and linked here.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
 changelog.html<br/>
 core/src/main/resources/hudson/model/ListView/configure-entries.jelly<br/>
<a href="http://jenkins-ci.org/commit/jenkins/adfa9a30c5913563f0c7a9a6dbf826c6901ef043" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/adfa9a30c5913563f0c7a9a6dbf826c6901ef043</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-20327&#93;</span> When saving a list view, submit POST data with size proportional to the # of jobs selected for inclusion, not the number of options.<br/>
This can be done easily by setting json="true"; otherwise the default is to pass =false for each unselected job.<br/>
Ironically, even with this fix we are sending 2× what is needed,<br/>
since currently ListView.submit does not even check the json parameter for job names:<br/>
it only pays attention to top-level request parameters that the browser defines for selected checkboxes.<br/>
Ideally we could ask buildFormTree to ignore a given form element entirely, so as to avoid the wasted data.</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/3170a1bc153b...adfa9a30c591" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/3170a1bc153b...adfa9a30c591</a></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
 changelog.html<br/>
 core/src/main/resources/hudson/model/ListView/configure-entries.jelly<br/>
<a href="http://jenkins-ci.org/commit/jenkins/db26457e49ee5cb740b3fe2dd44477377dff7f7e" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/db26457e49ee5cb740b3fe2dd44477377dff7f7e</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-20327&#93;</span> When saving a list view, submit POST data with size proportional to the # of jobs selected for inclusion, not the number of options.<br/>
This can be done easily by setting json="true"; otherwise the default is to pass =false for each unselected job.<br/>
Ironically, even with this fix we are sending 2× what is needed,<br/>
since currently ListView.submit does not even check the json parameter for job names:<br/>
it only pays attention to top-level request parameters that the browser defines for selected checkboxes.<br/>
Ideally we could ask buildFormTree to ignore a given form element entirely, so as to avoid the wasted data.<br/>
(cherry picked from commit adfa9a30c5913563f0c7a9a6dbf826c6901ef043)</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/4f89f983c154...db26457e49ee" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/4f89f983c154...db26457e49ee</a></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mboschetti">mboschetti</a>:</i>
<p>Sorry, but the problem is still there...<br/>
I obtain the following error</p>

<p>javax.servlet.ServletException: java.lang.IllegalStateException: Form too large 595864>200000</p>

<p>when I click Save to submit changes to even 1 job using Configuration Slicing plugin (http://<myjenkinshost>/slicing/jobdisabledbool/sliceconfigSubmit)</p>

<p>Jenkins version: 1.584<br/>
Configuration Slicing Plugin version: 1.39</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mwebber">mwebber</a>:</i>
<p>@marco, you should open a separate issue for the job config slicing plugin (this issue is for the core problem).<br/>
Also, if you read the comments above, you will find a workaround.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=leedega">leedega</a>:</i>
<p>Yes, the problem is related to submitting changes to view configurations.</p>

<p>Was there some specific information you'd like me to provide that hasn't already been stated above?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<blockquote><p>Was there some specific information you'd like me to provide that hasn't already been stated above?</p></blockquote>

<p>Is this a view with a very large number of items selected via checkbox?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=leedega">leedega</a>:</i>
<p>In my latest case the view was a "Sectioned View" type, with maybe 10 sub-sections of type "List View". Each list view subsection has a regular expression for the job filter, and there are few if any jobs that are checked off individually.</p>

<p>I suspect I can reproduce the problem on other view types as well. If I get some time today to try that out I'll let you know the results.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=leedega">leedega</a>:</i>
<p>I decided to just take some time right away to do some further tests and I can't seem to reproduce the problem on other view types. I have tried the following:</p>
<ul>
	<li>nested views</li>
	<li>list views</li>
	<li>categorized jobs view</li>
	<li>build pipeline view</li>
	<li>my view</li>
	<li>status view</li>
	<li>dashboard view</li>
</ul>


<p>Also, for some of our views of type "Sectioned View" with a smaller number of sub-sections on them, the error doesn't appear to occur. The combination of a view of type "Sectioned View" with a large number of sub-sections of type "List View" seems to be the problematic case. Also, our build farm has a large number of jobs on it (>1200 atm). If the Sectioned View were submitting information about each of these jobs for each of the list sub-views that could explain why the error only happens in this one use case.</p>

<p>Either way, the "hack" mentioned above for overriding the maxFormContentSize property for Jetty seems to prevent the problem, at least for us. This at least unblocks us for the time being. Still, it would be good to try and get a fix for this if possible.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<blockquote><p>In my latest case the view was a "Sectioned View" type, with maybe 10 sub-sections of type "List View". Each list view subsection has a regular expression for the job filter, and there are few if any jobs that are checked off individually.</p></blockquote>

<p>Please file an issue against Sectioned View Plugin to investigate whether the form size can be reduced there.</p>

<blockquote><p>the "hack" mentioned above for overriding the maxFormContentSize property for Jetty seems to prevent the problem</p></blockquote>

<p>Not really a hack, just a legitimate option. It's reasonable to limit the size of submitted forms to a sensible value by default, but larger Jenkins instances may easily grow past that, especially given how Jenkins performs form submissions.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=leedega">leedega</a>:</i>
<blockquote><p>Please file an issue against Sectioned View Plugin to investigate whether the form size can be reduced there.</p></blockquote>
<p>Done: <a href="https://issues.jenkins.io/browse/JENKINS-28716" title="Form Too Large error when changing view options" class="issue-link" data-issue-key="JENKINS-28716"><del>JENKINS-28716</del></a></p>

<blockquote><p>It's reasonable to limit the size of submitted forms to a sensible value by default...</p></blockquote>
<p>Agreed. The only debatable point is what value is 'sensible' in this context. In our context / use case it is obviously not sensible.</p>

<blockquote><p>Not really a hack, just a legitimate option.</p></blockquote>
<p>I'm not sure I agree with your assessment. As a user of the Jenkins product I expect the tool to work right out of the box with minimal manual modification / tweaking. Unfortunately, in this case, this was not possible. To make matters worse the bug itself was not obvious at first glance, and the root cause and workaround to the problem were also not directly obvious from the error produced. In fact, the only reason I knew of the existence of this maxFormContentSize option was by searching through bug reports until I came across this one which made mention of it. Hardly a simple process.</p>

<p>Conversely, I suspect that if you were to select a sufficiently large form size limit such that large scale users such as us wouldn't encounter this problem in the first place it would have little to no effect on the other users of the tool. That would be a preferable solution than having your users manually modify configuration files to make the tool work. Hence my assessment that this is in fact a 'hack'.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=leedega">leedega</a>:</i>
<p>PS: Sorry for the rant. I have been spending days / weeks testing the latest LTS edition of Jenkins in preparation for an upgrade to our build farm and I've been dealing with bug after bug, which are supposed to be 'resolved' by hack after hack and it has just been wearing on me. This experience is unfortunately reducing my confidence in the tool as a whole, and in the LTS maintenance stream in particular. <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>We chose to use the LTS edition instead of the head / master version to avoid problems just like this.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>In some sense, it's similar to configuring Xmx/XX:MaxPermSize in that it really depends on the size of the Jenkins instance what a reasonable value is, and it's up to you to increase it if your instance grows large. It's different as it should be easier to default to a large value without negative effects, but it won't really solve the problem &#8211; large requests will still happen depending on plugin.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=leedega">leedega</a>:</i>
<blockquote><p>It's different as it should be easier to default to a large value without negative effects..</p></blockquote>
<p>Yeah - that's what I suspected.</p>

<blockquote><p>...large requests will still happen depending on plugin.</p></blockquote>
<p>Yeah - this is I guess one of the unfortunate down-sides to the modular / plugin design of Jenkins. When something like this form size limit changes it takes time for that news to propagate out to all of the different maintainers of the different plugins, and even longer for them all to fix their plugins to compensate for the change. <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olivergondza">olivergondza</a>:</i>
<p>This is not a valid LTS candidate - no code changes to backport.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in jenkins<br/>
User: Stephen Connolly<br/>
Path:<br/>
 src/java/winstone/Launcher.java<br/>
 src/java/winstone/cmdline/Option.java<br/>
<a href="http://jenkins-ci.org/commit/winstone/9ae4206d85863b1f478bf21aa1a80a9111c66722" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/winstone/9ae4206d85863b1f478bf21aa1a80a9111c66722</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-20327&#93;</span><span class="error">&#91;FIXED JENKINS-23221&#93;</span><span class="error">&#91;FIXED JENKINS-24804&#93;</span><span class="error">&#91;FIXED JENKINS-21064&#93;</span> java.lang.IllegalStateException: Form too large</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in jenkins<br/>
User: Stephen Connolly<br/>
Path:<br/>
 src/java/winstone/Launcher.java<br/>
 src/java/winstone/cmdline/Option.java<br/>
<a href="http://jenkins-ci.org/commit/winstone/4a97150de06b6288e021cc191ae4189e4a21b3a3" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/winstone/4a97150de06b6288e021cc191ae4189e4a21b3a3</a><br/>
Log:<br/>
  Merge pull request #20 from stephenc/master</p>

<p><span class="error">&#91;FIXED JENKINS-20327&#93;</span><span class="error">&#91;FIXED JENKINS-23221&#93;</span><span class="error">&#91;FIXED JENKINS-24804&#93;</span>[FIXED JE...</p>


<p>Compare: <a href="https://github.com/jenkinsci/winstone/compare/f42497acd05d...4a97150de06b" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/winstone/compare/f42497acd05d...4a97150de06b</a></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick">jglick</a>:</i>
<p>I have marked a bunch of duplicates, but left them open, since it is typically still a crappy design in plugins (or core, as in this original issue) that they encourage forms with gigantic data. Jetty is now configured to meekly accept the whole submission, but your performance will still suffer.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mdonohue">mdonohue</a>:</i>
<p>What version of Jenkins picked up the jetty change to accept the whole submission?   I see Stephen Connolly's patch was applied to winstone, but there hasn't been a deploy of winstone with that patch based on the tags I see, and core/war/pom.xml points at winstone-2.8, which doesn't have the patch.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick">jglick</a>:</i>
<p>Indeed.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
 war/pom.xml<br/>
<a href="http://jenkins-ci.org/commit/jenkins/8f012d63d771b22d74eec8396b2895f0e45f0e6f" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/8f012d63d771b22d74eec8396b2895f0e45f0e6f</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-20327&#93;</span> Integrate Winstone 2.9.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
 changelog.html<br/>
 war/pom.xml<br/>
<a href="http://jenkins-ci.org/commit/jenkins/d76ee3e5179dd50e96e6c8e1d67ffbdec7f88a7b" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/d76ee3e5179dd50e96e6c8e1d67ffbdec7f88a7b</a><br/>
Log:<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-20327" title="“Form too large” errors submitting view configurations with many jobs" class="issue-link" data-issue-key="JENKINS-20327"><del>JENKINS-20327</del></a> Noting merge of #1922.</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/ab0198246c1c...d76ee3e5179d" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/ab0198246c1c...d76ee3e5179d</a></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dogfood">dogfood</a>:</i>
<p>Integrated in <span class="image-wrap" style=""><img src="https://ci.jenkins-ci.org/images/16x16/blue.png" style="border: 0px solid black" /></span> <a href="https://ci.jenkins-ci.org/job/jenkins_main_trunk/4373/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #4373</a><br/>
     <span class="error">&#91;FIXED JENKINS-20327&#93;</span> Integrate Winstone 2.9. (Revision 8f012d63d771b22d74eec8396b2895f0e45f0e6f)</p>

<p>     Result = SUCCESS<br/>
jesse glick : <a href="https://github.com/jenkinsci/jenkins/commit/8f012d63d771b22d74eec8396b2895f0e45f0e6f" class="external-link" target="_blank" rel="nofollow noopener">8f012d63d771b22d74eec8396b2895f0e45f0e6f</a><br/>
Files : </p>
<ul>
	<li>war/pom.xml</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
 war/pom.xml<br/>
<a href="http://jenkins-ci.org/commit/jenkins/1a91527767f6f4813835335506100848007623d6" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/1a91527767f6f4813835335506100848007623d6</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-20327&#93;</span> Integrate Winstone 2.9.</p>

<p>(cherry picked from commit 8f012d63d771b22d74eec8396b2895f0e45f0e6f)</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
<a href="http://jenkins-ci.org/commit/jenkins/e3ccde03907f3464dec56f5dab18e849ce508548" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/e3ccde03907f3464dec56f5dab18e849ce508548</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-20327&#93;</span> Integrate Winstone 2.9.</p>

<p>(cherry picked from commit 8f012d63d771b22d74eec8396b2895f0e45f0e6f)</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/f5f777f680b7" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/f5f777f680b7</a>^...e3ccde03907f</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dogfood">dogfood</a>:</i>
<p>Integrated in <span class="image-wrap" style=""><img src="https://ci.jenkins-ci.org/images/16x16/blue.png" style="border: 0px solid black" /></span> <a href="https://ci.jenkins-ci.org/job/jenkins_2.0/5/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_2.0 #5</a><br/>
     <span class="error">&#91;FIXED JENKINS-20327&#93;</span> Integrate Winstone 2.9. (Revision 8f012d63d771b22d74eec8396b2895f0e45f0e6f)</p>

<p>     Result = SUCCESS<br/>
jesse glick : <a href="https://github.com/jenkinsci/jenkins/commit/8f012d63d771b22d74eec8396b2895f0e45f0e6f" class="external-link" target="_blank" rel="nofollow noopener">8f012d63d771b22d74eec8396b2895f0e45f0e6f</a><br/>
Files : </p>
<ul>
	<li>war/pom.xml</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olivergondza">olivergondza</a>:</i>
<p>Not a candidate for 1.642 based LTS as it was fixed in 1.639.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-06-a2/issues?q=in%3Atitle%20JENKINS-24804">JENKINS-24804</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-06-a2/issues?q=in%3Atitle%20JENKINS-31521">JENKINS-31521</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Originally related to: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-06-a2/issues?q=in%3Atitle%20JENKINS-24421">JENKINS-24421</a>]</i>
